snapmaker/src/module/toolhead_laser.cpp

```diff
@@ -1025,16 +1025,34 @@ void ToolHeadLaser::InlineDisable() {
 
 void ToolHeadLaser::SetOutputInline(float power) {
   CheckFan(power);
-  planner.laser_inline.status.isEnabled = true;
 
   SetPower(power);
   planner.laser_inline.power = power_pwm_;
+  
+  if (power_pwm_ > 0)
+    planner.laser_inline.status.isEnabled = true;
+  else
+    planner.laser_inline.status.isEnabled = false;
+
+  if (laser->device_id_ == MODULE_DEVICE_ID_10W_LASER && power_pwm_ > 0) {
+    if (laser_10w_status_ == LASER_10W_DISABLE) {
+      // Send the enable command only when the power is disabled
+      LaserControl(1);
+    }
+    // Stop waiting disable
+    laser_10w_status_ = LASER_10W_ENABLE;
+  }
 }
 
 
 void ToolHeadLaser::TurnOn_ISR(uint16_t power_pwm) {
   if (power_pwm > power_limit_pwm_)
     power_pwm = power_limit_pwm_;
 
+  if (power_pwm > 0)
+    state_ = TOOLHEAD_LASER_STATE_ON;
+  else
+    state_ = TOOLHEAD_LASER_STATE_OFF;
```

Comment:
A method called "TurnOn" that turns off the laser is kinda confusing, we need to do this so we don't accidentally turn the laser back on when the enclosure is opened? (if so, maybe it'd be better to dig in why we're attempting to do `laser->TurnOn()` when `laser->SetPowerLimit()` is called, since that also has a code smell to me)

Reply:
> A method called "TurnOn" that turns off the laser is kinda confusing

This name I chose as an extension to the previously heavily used TurnOn(). Maybe a better name would be like ISR_Laser_Control or something.

Reply:
I think this has the same feeling as above (on and 0 vs off)